test/exe/main_common_test.cc

```diff
@@ -129,29 +129,50 @@ TEST_P(MainCommonTest, ConstructWritesBasePathId) {
 }
 
 // Exercise enabling core dump and succeeding.
-// Note: this test will call the real prctl(), which is what we want.
+// Note: this test will call the real system call, which is what we want.
 TEST_P(MainCommonTest, EnableCoreDump) {
+#ifdef __linux__
   addArg("--enable-core-dump");
-  VERBOSE_EXPECT_NO_THROW(MainCommon main_common(argc(), argv()));
+  auto test = [&]() { MainCommon main_common(argc(), argv()); };
+  EXPECT_LOG_CONTAINS("info", "core dump enabled", test());
+#endif
 }
 
 class MockPlatformImpl : public PlatformImpl {
-  bool enableCoreDump() override { return false; }
-}
+public:
+  MockPlatformImpl(bool& called) : called_(called) {}
+  bool enableCoreDump() override {
```

Comment:
nit: you can just add a MOCK_METHOD here that allows you to have a normal expectation, etc.?

Reply:
sure done